flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/PulsarSourceBuilder.java

```diff
@@ -422,15 +421,12 @@ public PulsarSource<OUT> build() {
         checkNotNull(subscriber, "No topic names or topic pattern are provided.");
 
         SubscriptionType subscriptionType = configBuilder.get(PULSAR_SUBSCRIPTION_TYPE);
-        if (subscriptionType == SubscriptionType.Key_Shared) {
-            if (rangeGenerator == null) {
+        if (rangeGenerator == null) {
+            if (subscriptionType == SubscriptionType.Key_Shared) {
                 LOG.warn(
                         "No range generator provided for key_shared subscription,"
                                 + " we would use the UniformRangeGenerator as the default range generator.");
```

Comment:
This warning is incorrect since we don't assign `UniformRangeGenerator` to `this.rangeGenerator` now?

Reply:
Yep. You are right.